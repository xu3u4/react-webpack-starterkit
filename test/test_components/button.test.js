import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react'

import Button from 'components/button';

describe ('<Button>', () => {
  const onclick = jest.fn();

  it ('should render button properly', () => {
    const value = 'Click';
    const { getByText } = render(
      <Button onClick={ onclick } >{ value }</Button>
    );

    expect(getByText(value)).toBeDefined()
  });

  it ('should render onClick when click', () => {
    const value = 'Click';
    const { getByText } = render(
      <Button onClick={ onclick } >{ value }</Button>
    );

    fireEvent.click(getByText('Click'))
    expect(onclick).toHaveBeenCalledTimes(1)
  });
});