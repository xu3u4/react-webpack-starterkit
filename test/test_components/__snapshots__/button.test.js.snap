// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> render components 1`] = `
<button
  className="simple_btn"
  onClick={[Function]}
  type="button"
>
  Click
</button>
`;
